src/App.jsx

```diff
@@ -14,45 +18,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT__ALPHABETICALLY = 'alphabet';
+const SORT__BY__LENGTH = 'length';
+const getPreparedGoods = (goods, sortType, isReversed) => {
+  const copyGoods = [...goods];
+
+  if (sortType) {
+    switch (sortType) {
+      case SORT__ALPHABETICALLY:
+        copyGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SORT__BY__LENGTH:
+        copyGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default: return 0;
+    }
+  }
+
+  return isReversed ? copyGoods.reverse() : copyGoods;
+};
+
+export const App = () => {
+  const [sortType, setsortType] = useState(null);
+  const [isReversed, setIsReversed] = useState('');
+  const preparedGoods = getPreparedGoods(
+    goodsFromServer, sortType, isReversed,
+  );
+
+  const handleReset = () => {
+    setsortType(null);
+    setIsReversed('');
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetCondition = (sortType !== null || isReversed === 'reverse');
```

Comment:
`const resetCondition = (sortType || isReversed)`. If you make `isReverse` boolean